src/main/java/org/apache/sling/api/redirect/RedirectResolver.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.sling.api.redirect;
+
+
+import org.osgi.annotation.versioning.ProviderType;
+
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * <p>
+ * A RedirectResolver resolves a redirect in the context of a request. It resolves that redirect having first been bound
+ * to a Resource. This is typically achieved through the Adapt to mechanism eg resource.adaptTo(RedirectResolver.class).
+ * </p><p>
+ * Implementations should implement an AdapterFactory to perform the adaption to an implementation of this class. That class
+ * should implement the resolve method such that the methods of the Redirect Response are called to define the RedirectResponse.
+ * </p><p>
+ * The implementation of the resolve method indicates that it has successfully resolved a redirect by setting a setting a status code
+ * see the documentation of the RedirectResponse class for more information.
+ * </p>
+ */
+@ProviderType
```

Comment:
This is a consumer type interface, its not provided by the resourceresolver impl

Reply:
The interface may be consumed by many services, not limited to the Sling Default GET servlet. eg, if something that writes html needs to pre-resolved the redirect, then it will call RedirectResolver.resolve(request, redirect); using the request generating the html as the request context.  If it was only the Sling Default GET servlet, then I agree it should be a ConsumerType, but there are many, hence a ProviderType (perhaps I got this wrong, not being an OSGi expert)

Reply:
I changed this to ConsumerType. If you want it changing back to provider type on the basis that it will likely be used in many components rendering html or anything else where a link is included, then please say.

Reply:
if an interface is implemented by many bundles, thats a good hint that is is a consumer type - so all fine here

Reply:
I guess I missed that discussion - I think this should be a ConsumerType, no?

Reply:
@karlpauls is right here - we expect implementations outside of the Sling codebase, either in a `ResourceProvider` or through an `AdapterFactory`. And it's the same analogy that @cziegeler suggested.

Reply:
These annotations are really easy to use :) But I like agreeing with myself, so ConsumerType is probably the way to go